framework/test/unit/modules/interoperability/sidechain/module.spec.ts

```diff
@@ -0,0 +1,546 @@
+				);
+			});
+
+			it(`should throw error if === ${CHAIN_NAME_MAINCHAIN}`, async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...defaultData,
+						ownChainName: CHAIN_NAME_MAINCHAIN,
+					},
+					params,
+				);
+
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`ownChainName must be not equal to ${CHAIN_NAME_MAINCHAIN}.`,
+				);
+			});
+		});
+
+		it('should throw error if not ownChainNonce > 0', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...defaultData,
+					ownChainNonce: BigInt(0),
+				},
+				params,
+			);
+
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				'ownChainNonce must be > 0.',
+			);
+		});
+
+		it('should throw error if chainInfos.length !== 1', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...defaultData,
+					chainInfos: [chainInfo, chainInfo],
+				},
+				params,
+			);
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				'chainInfos must contain exactly one entry.',
+			);
+		});
+
+		it('should throw error if mainchainInfo.chainID is not equal to getMainchainID()', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...defaultData,
+					chainInfos: [
+						{
+							...chainInfo,
+							chainID,
+						},
+					],
+				},
+				params,
+			);
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				`mainchainInfo.chainID must be equal to ${getMainchainID(chainID).toString('hex')}.`,
+			);
+		});
+
+		describe('chainInfo.chainData', () => {
+			it(`should throw error if chainData.name !== ${CHAIN_NAME_MAINCHAIN}`, async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...defaultData,
+						chainInfos: [
+							{
+								...defaultData.chainInfos[0],
+								chainData: {
+									...chainData,
+								},
+							},
+						],
+					},
+					params,
+				);
+
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`chainData.name must be equal to ${CHAIN_NAME_MAINCHAIN}.`,
+				);
+			});
+
+			it('should throw error if chainData.status is not CHAIN_STATUS_REGISTERED or CHAIN_STATUS_ACTIVE', async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...defaultData,
+						chainInfos: [
+							{
+								...defaultData.chainInfos[0],
+								chainData: {
+									...defaultData.chainInfos[0].chainData,
+									status: ChainStatus.TERMINATED,
+								},
+							},
+						],
+					},
+					params,
+				);
+
+				const validStatuses = [ChainStatus.REGISTERED, ChainStatus.ACTIVE];
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`chainData.status must be one of ${validStatuses.join(', ')}.`,
+				);
+			});
+
+			it('should throw error if chainData.lastCertificate.timestamp > g.header.timestamp', async () => {
```

Comment:
Should throw error if `chainData.lastCertificate.timestamp >= g.header.timestamp`. So I would add an extra test case where `chainData.lastCertificate.timestamp = g.header.timestamp` to make sure that it throws error in this case as well.

Reply:
Added, I am resolving.